src/AnimationManager.ts

```diff
@@ -128,35 +113,183 @@ export class AnimationManager {
+			const { range, circular } = opt;
+
+			if (circular && (circular[0] || circular[1])) {
+				return v;
+			} else {
+				return getInsidePosition(v, range, circular as boolean[]);
+			}
+		});
+
+		if (equal(movedPos, orgPos)) {
+			return this;
+		}
+
+		if (duration > 0) {
+			this.animateTo(movedPos, duration);
+		} else {
+			this.em.triggerChange(movedPos);
+			this.finish(false);
+		}
+
+		return this;
+	}
+
+	public setBy(pos: Axis, duration = 0) {
+		return this.setTo(
+			map(this.axm.get(Object.keys(pos)), (v, k) => v + pos[k]),
+			duration,
+		);
+	}
+
+	public updateAnimationPos(pos: Axis, restart: boolean): void {
+		const animateParam = this._animateParam;
+		if (!animateParam) {
+			return;
+		}
+		if (restart) {
+			this.setTo(pos, animateParam.duration - (new Date().getTime() - animateParam.startTime));
+		} else {
+			const currentPos = this.axm.get();
+			// When destination is changed, new delta should be calculated as remaining percent.
+			// For example, moving x:0, y:0 to x:200, y:200 and it has current easing percent of 92%. coordinate is x:184 and y:184
+			// If destination changes to x:300, y:300. xdelta:200, ydelta:200 changes to xdelta:116, ydelta:116 and use remaining easingPer as 100%, not 8% as previous.
+			// Therefore, original easingPer by time is kept. And divided by (1 - self._initialEasingPer) which means new total easing percent. Like calculating 8% as 100%.
+			this._initialEasingPer = this._prevEasingPer;
+			animateParam.delta = this.axm.getDelta(currentPos, pos);
+			animateParam.destPos = pos;
+		}
+	}
+
+	public updateAnimationDuration(duration: number,  restart: boolean): void {
+		const animateParam = this._animateParam;
+		if (!animateParam) {
+			return;
+		}
+		if (restart) {
+			this.setTo(animateParam.destPos, duration - (new Date().getTime() - animateParam.startTime));
+		} else {
+			const diffTime = new Date().getTime() - animateParam.startTime;
+			const ratio = (diffTime + this._durationOffset) / animateParam.duration;
+			// Use durationOffset for keeping animation ratio after duration is changed.
+			// newRatio = (diffTime + newDurationOffset) / newDuration = oldRatio
+			// newDurationOffset = oldRatio * newDuration - diffTime
+			this._durationOffset = ratio * duration - diffTime;
+			animateParam.duration = duration;
+		}
+	}
+
+	private createAnimationParam(pos: Axis, duration: number, option?: ChangeEventOption): AnimationParam {
+		const depaPos: Axis = this.axm.get();
+		const destPos: Axis = pos;
+		const inputEvent = option && option.event || null;
+		return {
+			depaPos,
+			destPos,
+			duration: minMax(
+				duration,
+				this._options.minimumDuration,
+				this._options.maximumDuration),
+			delta: this.axm.getDelta(depaPos, destPos),
+			inputEvent,
+			input: option && option.input || null,
+			isTrusted: !!inputEvent,
+			done: this.animationEnd,
+		};
+	}
+
+	private animateLoop(param: AnimationParam, complete: () => void): void {
```

Comment:
Did you set the eslint properly? Looks like the underscore prefix is missing for those private methods.